helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -434,6 +443,9 @@ service:
         {{- end }}
         - k8sattributes
         - batch
+        {{- if .Values.logsCollection.containers.useSplunkIncludeAnnotation }}
+        - filter/include_pod_logs
+        {{- end }}
```

Comment:
This should go under the next condition `{{- if not .Values.otelCollector.enabled }}`. Otherwise, with `otelCollector=enable`, it won't do anything because there is no k8s metatada available at this point to filter the logs on.

Another question. Do we still need to handle `splunk.com/exclude` if `logsCollection.containers.useSplunkIncludeAnnotation=true`? 

If not, we can avoid setting up unnecessary processors and reduce amount of code by just changing the helper for `filter/logs`:

```
filter/logs:
  logs:
    {{ .Values.logsCollection.containers.useSplunkIncludeAnnotation | "include" "exclude" }}:
      match_type: strict
      resource_attributes:
        - key: splunk.com/{{ .Values.logsCollection.containers.useSplunkIncludeAnnotation | "include" "exclude" }}
          value: "true"
```

What do you think?

Reply:
really good idea, i think we should just use your definition of filter/logs